apps/site/assets/ts/helpers/service.ts

```diff
@@ -1,155 +1,197 @@
+    if (service.typicality === "holiday_service") {
+      return "Holiday Schedules";
+    }
 
-  const startDateObject = new Date(startDate);
-  const endDateObject = new Date(endDate);
-  if ("service_date" in service) {
-    // Get unix timestamps
-    const serviceDateTime = new Date(service.service_date).getTime();
-    const startDateTime = startDateObject.getTime();
-    const endDateTime = endDateObject.getTime();
+    // if there's no rating end date, this is probably a CR or Ferry.
+    // don't use the rating in that case.
+    if (hasNoRating(service)) {
+      if (isInCurrentService(service, currentDate)) {
+        return "Current Schedules";
+      }
+      if (isInFutureService(service, currentDate)) {
+        return "Future Schedules";
+      }
+    } else {
+      const ratingStartDate = new Date(service.rating_start_date!);
+      const ratingEndDate = new Date(service.rating_end_date!);
+
+      if (
+        isInCurrentRating(service, currentDate) &&
+        service.typicality === "typical_service"
+      ) {
+        return `Current Schedules (${
+          service.rating_description
+        }, ends ${shortDate(ratingEndDate)})`;
+      }
 
-    if (serviceDateTime >= startDateTime && serviceDateTime <= endDateTime) {
-      return `ends ${formattedDate(endDateObject)}`;
+      if (isInFutureRating(service, currentDate)) {
+        return `Future Schedules (${
+          service.rating_description
+        }, starts ${shortDate(ratingStartDate)})`;
+      }
     }
 
-    if (serviceDateTime < startDateTime) {
-      return `starts ${formattedDate(startDateObject)}`;
-    }
-  }
-  return startToEnd(startDateObject, endDateObject);
+    return "Other Schedules";
+  });
+
+// sort the option groups
+const optGroupOrder: { [key: string]: number } = {
+  Current: 4,
+  Holiday: 3,
+  Future: 2,
+  Other: 1
 };
 
-type MonthInteger = number | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+export const optGroupComparator = (groupA: string, groupB: string): number => {
+  const a = groupA.split(" ")[0];
+  const b = groupB.split(" ")[0];
+  return optGroupOrder[b] - optGroupOrder[a];
+};
 
-export const hasMultipleWeekdaySchedules = (
-  services: ServiceWithServiceDate[]
-): boolean =>
+export const hasMultipleWeekdaySchedules = (services: Service[]): boolean =>
   services.filter(
     service =>
       service.type === "weekday" && service.typicality !== "holiday_service"
   ).length > 1;
+
+const dayIntegerToString = (day: DayInteger): string =>
+  [
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+    "Sunday"
+  ][day - 1];
+
+const daysAreConsecutive = (
+  days: DayInteger[],
+  areConsecutive: boolean
+): boolean => {
+  if (!areConsecutive) return areConsecutive;
+  const [day, nextDay, ...rest] = days;
+  if (rest.length === 0) return day + 1 === nextDay;
+  return daysAreConsecutive([nextDay, ...rest], day + 1 === nextDay);
```

Comment:
I think we could express the same logic without needing an extra argument on the function:

```js
const daysAreConsecutive = (days: DayInteger[]): boolean => {
  const [day, nextDay, ...rest] = days;
  if (rest.length === 0) return day + 1 === nextDay;
  if (day + 1 !== nextDay) return false;
  return daysAreConsecutive([nextDay, ...rest]);
};
```

Reply:
Good call, I think that argument was an artifact of how this function was invoked previously. I'll adjust that